File: src/main/java/com/learnershi/rclasssocket/repository/CanvasDrawRepository.java
package com.learnershi.rclasssocket.repository;

import com.learnershi.rclasssocket.entity.CanvasDraw;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Flux;

public interface CanvasDrawRepository extends ReactiveMongoRepository<CanvasDraw, String> {
    Flux<CanvasDraw> findAllByClassRoomId(String classRoomId);

    Flux<CanvasDraw> findAllByClassRoomIdAndUserSeqAndTabIndexAndPageIndex(String classRoomId, String userSeq, Integer tabIndex, Integer pageIndex);

    Flux<CanvasDraw> findAllByClassRoomIdAndUserSeqAndTabIndex(String classRoomId, String userSeq, Integer tabIndex);
}

File: src/main/java/com/learnershi/rclasssocket/repository/CanvasDrawRepository.kt
package com.learnershi.rclasssocket.repository

import com.learnershi.rclasssocket.entity.CanvasDraw
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux

interface CanvasDrawRepository : ReactiveMongoRepository<CanvasDraw?, String?> {
    fun findAllByClassRoomId(classRoomId: String?): Flux<CanvasDraw?>
    fun findAllByClassRoomIdAndUserSeqAndTabIndexAndPageIndex(
        classRoomId: String?,
        userSeq: String?,
        tabIndex: Int?,
        pageIndex: Int?
    ): Flux<CanvasDraw?>?

    fun findAllByClassRoomIdAndUserSeqAndTabIndex(
        classRoomId: String?,
        userSeq: String?,
        tabIndex: Int?
    ): Flux<CanvasDraw?>?
}